Sources/SwiftJWT/JWTEncoder.swift

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Foundation
+import KituraContracts
+
+// MARK: JWTEncoder
+
+/**
+ A thread safe encoder that signs the JWT header and claims using the provided algorithm and encodes a `JWT` instance as either Data or a JWT String.
+ 
+ ### Usage Example: ###
+ ```swift
+ struct MyClaims: Claims {
+    var name: String
+ }
+ var jwt = JWT(claims: MyClaims(name: "John Doe"))
+ let rsaPrivateKey = read(fileName: "rsa_private_key")
+ let rsaJWTEncoder = JWTEncoder(algorithm: Algorithm.rs256(rsaPrivateKey, .privateKey))
+ do {
+    let jwtString = try rsaJWTEncoder.encodeToString(jwt)
+ } catch {
+    print("Failed to encode JWT: \(error)")
+ }
+ ```
+ */
+public class JWTEncoder: BodyEncoder {
+    
+    let keyIDToSigner: (String) -> JWTSigner?
+    let jwtSigner: JWTSigner?
+    let header: Header?
+    
+    // MARK: Initializers
+    
+    /// Initialize a `JWTEncoder` instance with a single `JWTSigner`.
+    ///
+    /// - Parameter algorithm: The `Algorithm` that will be used to sign the JWT.
+    /// - Returns: A new instance of `JWTEncoder`.
+    public init(jwtSigner: JWTSigner, header: Header? = nil) {
```

Comment:
I find this concept of a default `Header` a bit strange.  I understand that this is so that you could use it to encode a `Claims` object which is not a `JWT` with its own Header.  What is the use case for using a JWTEncoder in this way?